RELEASE_PROCESS.md

```diff
@@ -124,18 +98,7 @@ The git SHA of the build you are testing can be found in
 
 ### Building and Releasing the Release
 
-- [ ] Tag the chef-server repository with the release version: `git
-  tag -a VERSION_NUMBER`. The first line of the tag message should be
-  the version number. The other lines are up to you. You're the boss!
-  (You can leave them blank).
-
-- [ ] Push the new tag: `git push origin master --tags`.
-
-- [ ] Trigger a release build in Jenkins using the
-  `chef-server-12-trigger-release` trigger.  Use the tag you created
-  above as the GIT_REF parameter.
-
-- [ ] Wait for the pipeline to complete.
+- [ ] Select a version from the `current` channel that you wish to promote to `stable`.
```

Comment:
Might be good to make it clear that we should promote the same version we sent through the upgrade tests.